File: etc/nixos/config/herbstluftwm/python/startup.py
import os

# ----- ----- ----- ----- ----- ----- ----- ----- ----- ----- ----- ----
# load on startup

def run():
    command = 'silent new_attr bool my_not_first_autostart'
    exitcode = os.system('herbstclient ' + command)

    if exitcode == 0:
      # non windowed app
        os.system("systemctl --user restart picom &")
        os.system("dunst &")
        os.system("parcellite &")
        os.system("sh ~/.fehbg")
        os.system("sh ~/.config/polybar/launch.sh")
        # os.system("sh ~/.config/herbstluftwm/scripts/lmms.sh")
        os.system("mpd &")

      # windowed app
       # os.system("xfce4-terminal &")
       # os.system("sleep 1 && firefox &")
       # os.system("sleep 2 && geany &")
       # os.system("sleep 2 && thunar &")

File: root/config/herbstluftwm/lua/startup.lua
local helper  = require('.helper')

local _M = {}

-- ----- ----- ----- ----- ----- ----- ----- ----- ----- ----- -----
-- load on startup

function _M.run()
    -- redirect stderror to stdout, then capture the result
    command = 'new_attr bool my_not_first_autostart'
    local handle = io.popen('herbstclient ' .. command .. ' 2>&1')
    local result = handle:read('*a')
    local exitcode = handle:close()
    
    if ((result == nil or result == '')) then
     -- non windowed app
        os.execute('compton &')
        os.execute('dunst &')
        os.execute('parcellite &')
        os.execute('nitrogen --restore &')
        os.execute('mpd &')

     -- windowed app
        os.execute('xfce4-terminal &')
        os.execute('sleep 1 && firefox &')
        os.execute('sleep 2 && geany &')
        os.execute('sleep 2 && thunar &')
    end
end

-- ----- ----- ----- ----- ----- ----- ----- ----- ----- ----- -----
-- return

return _M
